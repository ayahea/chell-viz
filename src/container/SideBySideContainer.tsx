import * as React from 'react';
import { Dropdown, Grid, GridColumn, GridRow } from 'semantic-ui-react';

import { SpringContainer } from './SpringContainer';
import { TContainer } from './TContainer';

export interface ISideBySideContainerState {
  currentDataDir: string;
  dataDirs: string[];
}

export class SideBySideContainer extends React.Component<any, ISideBySideContainerState> {
  public constructor(props: any) {
    super(props);
    this.state = {
      currentDataDir: 'centroids_subset',
      dataDirs: ['centroids', 'centroids_subset', 'spring2/full'],
    };
  }

  public render() {
    return (
      <Grid>
        <GridRow columns={1} centered={true}>
          <Dropdown
            onChange={this.onDataDirChange}
            options={[
              ...this.state.dataDirs.map(dir => {
                return { key: dir, text: dir, value: dir };
              }),
            ]}
            placeholder={'Select Data Directory'}
            search={true}
          />
        </GridRow>
        <GridRow columns={2}>
          <GridColumn>
            <TContainer dataDir={this.state.currentDataDir} />
          </GridColumn>
          <GridColumn>
            <SpringContainer dataDir={this.state.currentDataDir} />
          </GridColumn>
        </GridRow>
      </Grid>
    );
  }

  protected onDataDirChange = (event: React.SyntheticEvent<any>, data: any) => {
    this.setState({
      currentDataDir: data.value,
    });
  };
}
