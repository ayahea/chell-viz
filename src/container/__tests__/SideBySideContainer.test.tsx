import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';

import { SideBySideContainer } from '../SideBySideContainer';

describe('SideBySideContainer', () => {
  test('Should match existing snapshot when given no data.', () => {
    expect(toJson(shallow(<SideBySideContainer />))).toMatchSnapshot();
  });
});
