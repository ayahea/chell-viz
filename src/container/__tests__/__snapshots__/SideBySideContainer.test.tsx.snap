// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBySideContainer Should match existing snapshot when given no data. 1`] = `
<Grid>
  <GridRow
    centered={true}
    columns={1}
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      deburr={false}
      icon="dropdown"
      minCharacters={1}
      noResultsMessage="No results found."
      onChange={[Function]}
      openOnFocus={true}
      options={
        Array [
          Object {
            "key": "centroids",
            "text": "centroids",
            "value": "centroids",
          },
          Object {
            "key": "centroids_subset",
            "text": "centroids_subset",
            "value": "centroids_subset",
          },
          Object {
            "key": "spring2/full",
            "text": "spring2/full",
            "value": "spring2/full",
          },
        ]
      }
      placeholder="Select Data Directory"
      renderLabel={[Function]}
      search={true}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      wrapSelection={true}
    />
  </GridRow>
  <GridRow
    columns={2}
  >
    <GridColumn>
      <TContainer
        dataDir="centroids_subset"
      />
    </GridColumn>
    <GridColumn>
      <SpringContainer
        dataDir="centroids_subset"
      />
    </GridColumn>
  </GridRow>
</Grid>
`;
